contracts/nois-sink/src/contract.rs

```diff
@@ -87,6 +88,48 @@ fn execute_burn(deps: DepsMut, info: MessageInfo, env: Env) -> Result<Response,
         .add_attribute("time", time.to_string()))
 }
 
+fn execute_burn_native(
+    deps: DepsMut,
+    info: MessageInfo,
+    env: Env,
+) -> Result<Response, ContractError> {
+    if !info.funds.is_empty() {
+        return Err(ContractError::NonPayableMessage);
+    }
+    let contract_balance = deps
+        .querier
+        .query_balance(&env.contract.address, BURN_DENOM)?;
+
+    if contract_balance.amount.is_zero() {
+        return Err(ContractError::NoFundsToBurn);
+    }
+
+    let new_id = ASHES_LAST_ID.may_load(deps.storage)?.unwrap_or_default() + 1;
+    ASHES_LAST_ID.save(deps.storage, &new_id)?;
+
+    let time = env.block.time;
+
+    //store the burner Ash
+    ASHES.save(
+        deps.storage,
+        new_id,
+        &Ash {
+            burner: None,
+            amount: contract_balance.to_owned(),
```

Comment:
This is owned already. Better use clone, which does the same here but better expresses the intend:

```suggestion
            amount: contract_balance.clone(),
```